engine/src/flutter/impeller/geometry/matrix.cc

```diff
@@ -358,6 +360,76 @@ std::optional<MatrixDecomposition> Matrix::Decompose() const {
   return result;
 }
 
+std::optional<std::pair<Scalar, Scalar>> Matrix::GetScales2D() const {
```

Comment:
Couldn't you just transform 2 unit vectors to calculate this?

Reply:
No, this isn't the scale factor for the unit vectors - that's what the basis methods return. This is the expansion factor in any direction. For skewed transforms, the unit axis vectors are not scaled, but the 45-degree unit vector is scaled by more than 1, and the perpendicular to that vector is scaled by less than 1.